table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1470,82 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listMetadataChangedTables", absolute = true)
+    @Override
+    public Iterator<ChangedTable> listMetadataChangedTables(DateTime fromInclusive, DateTime toExclusive, LimitCounter limit) {
+        checkArgument(limit.remaining() > 0, "Limit must be > 0");
+        checkArgument(fromInclusive != null, "from date cannot be null");
+        checkArgument(toExclusive != null, "to date cannot be null");
+
+        LocalDate fromDate = LocalDate.of(fromInclusive.withZone(DateTimeZone.UTC).getYear(), fromInclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                fromInclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
+        LocalDate toDate = LocalDate.of(toExclusive.withZone(DateTimeZone.UTC).getYear(), toExclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                toExclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
```

Comment:
This conversion loses any time information.  Why can't the caller ask for all changes between say, 2018-01-01 08:00 and 2018-01-01 16:00?

I'd recommend changing to the following logic:

1. Identify the UTC start-of-day for `fromInclusive`.
2. If `toExclusive` is after midnight then use the UTC start of day for `toExclusive` as your final date, otherwise use the previous day.
3. Iterate inclusively over all the dates between the dates from 1) and 2).
4. For each event returned filter events whose time is outside the range of `[fromInclusive, toExclusive)`.